app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileViewModel.kt

```diff
@@ -78,18 +79,26 @@ class SelfUserProfileViewModel @Inject constructor(
 
     private fun observeEstablishedCall() {
         viewModelScope.launch {
-            observeEstablishedCalls().map { it.isNotEmpty() }.distinctUntilChanged().collect {
-                userProfileState = userProfileState.copy(isUserInCall = it)
-            }
+            observeEstablishedCalls()
+                .map { it.isNotEmpty() }
+                .distinctUntilChanged()
+                .flowOn(dispatchers.io())
+                .shareIn(this, SharingStarted.WhileSubscribed(1))
+                .collect {
+                    userProfileState = userProfileState.copy(isUserInCall = it)
+                }
         }
     }
 
     private suspend fun fetchSelfUser() {
         viewModelScope.launch {
+            val self = getSelf().flowOn(dispatchers.io()).shareIn(this, SharingStarted.WhileSubscribed(1))
```

Comment:
same here `viewModelScope.launch(dispatcher.io())` and no need in `.flowOn(dispatchers.io())`

Reply:
@borichellow in the collect block the SelfUserProfileState is updated and compose state can only be updated from the main thread. Otherwise, it can lead to runtime exception.

it's better to change the Use cases to be safe to lunch from main instead of switching the dispatcher in the view model.